src/hooks/use-reach-errors.ts

```diff
@@ -0,0 +1,36 @@
+import { APIManager } from "@stakekit/api-hooks";
+import { signal, useSignalEffect } from "@preact/signals-react";
+import { useState } from "react";
+import { ErrorsSet, errorsSet } from "../utils/errors";
+
+export const reachErrorKey = signal<ReachError | undefined>(undefined);
```

Comment:
Isn't this the same thing as storing state in context and exposing it globally, or signal provides some other benefits here I'm not aware? If global scope is needed, lets use context for this

Reply:
You are right, the first version of the code used signal directly without any react hooks, but after introducing a useReachError hook as a wrapper, the Signal stopped making any sense.